CliClient/tests/services_rest_Api.js

```diff
@@ -326,4 +327,50 @@ describe('services_rest_Api', function() {
 		expect(response3.length).toBe(2);
 	}));
 
+	it('should update tags when updating notes', asyncTest(async () => {
+		const tag1 = await Tag.save({ title: 'mon étiquette 1' });
+		const tag2 = await Tag.save({ title: 'mon étiquette 2' });
+		const tag3 = await Tag.save({ title: 'mon étiquette 3' });
+		const newTagTitle = 'mon étiquette 4';
+
+		const note = await Note.save({
+			title: 'ma note un',
+		});
+		Tag.addNote(tag1.id, note.id);
+		Tag.addNote(tag2.id, note.id);
+
+		const response1 = await api.route('PUT', `notes/${note.id}`, null, JSON.stringify({
+			tags: `${tag1.title},${tag3.title}`,
+		}));
+		const tagIds1 = await NoteTag.tagIdsByNoteId(note.id);
+		expect(response1.tags === `${tag1.title},${tag3.title}`).toBe(true);
+		expect(tagIds1.length === 2).toBe(true);
+		expect(tagIds1.includes(tag1.id)).toBe(true);
+		expect(tagIds1.includes(tag3.id)).toBe(true);
+
+		const response2 = await api.route('PUT', `notes/${note.id}`, null, JSON.stringify({
+			tags: `${tag2.title},${newTagTitle}`,
+		}));
+		const newTag = await Tag.loadByTitle(newTagTitle);
+		const tagIds2 = await NoteTag.tagIdsByNoteId(note.id);
+		expect(response2.tags === `${tag2.title},${newTag.title}`).toBe(true);
+		expect(tagIds2.length === 2).toBe(true);
+		expect(tagIds2.includes(tag2.id)).toBe(true);
+		expect(tagIds2.includes(newTag.id)).toBe(true);
+
+		const response3 = await api.route('PUT', `notes/${note.id}`, null, JSON.stringify({
+			title: 'Some other title',
+		}));
+		const tagIds3 = await NoteTag.tagIdsByNoteId(note.id);
+		expect(tagIds3.length === 2).toBe(true);
+		expect(tagIds3.includes(tag2.id)).toBe(true);
+		expect(tagIds3.includes(newTag.id)).toBe(true);
+
+		const response4 = await api.route('PUT', `notes/${note.id}`, null, JSON.stringify({
+			tags: '',
+		}));
+		const tagIds4 = await NoteTag.tagIdsByNoteId(note.id);
+		expect(response4.tags === '').toBe(true);
+		expect(tagIds4.length === 0).toBe(true);
+	}));
```

Comment:
Generally, for future maintenance, it would be better to separate the test cases into 3 or 4 smaller test cases rather than one big one. This also helps avoid using the 1,2,3,4 variable names too.  Sometimes it is too inefficient, but by default it is preferred.

Reply:
I restructured all the test cases into separate ones. But, I'm not sure if I did it right. I have used lengthy test case descriptions for some test cases. Please review and let me know if I'm doing it right.

Reply:
Looks good!